web/src/app/profile/approved-researcher/components/AgreementText.tsx

```diff
@@ -2,31 +2,11 @@
 
 import "./AgreementText.css";
 import Markdown from "react-markdown";
-import { useEffect, useState } from "react";
-import { getAgreementsApprovedResearcher } from "@/openapi";
-
-export default function AgreementText() {
-  const [agreementText, setAgreementText] = useState<string | null>(null);
-
-  useEffect(() => {
-    const fetchAgreement = async () => {
-      try {
-        const res = await getAgreementsApprovedResearcher();
-
-        if (res.response.status === 200 && res.data) {
-          setAgreementText(res.data.text);
-        }
-      } catch (err) {
-        console.error("Agreement fetch error:", err);
-      }
-    };
-
-    fetchAgreement();
-  }, []);
 
+export default function AgreementText({ text }: { text: string }) {
   return (
     <section className="section">
-      <Markdown>{agreementText}</Markdown>
+      <Markdown>{text}</Markdown>
     </section>
   );
 }
```

Comment:
Here is an alternative syntax for this:

```
type AgreementTextProps = {
  agreementText: string;
};

export default function AgreementText(props: AgreementTextProps) {
  const { agreementText } = props;

  return (
    <section className="section">
      <Markdown>{agreementText}</Markdown>
    </section>
  );
}

```

It is more verbose, and more of a personal preference. But it does make it easier to read if your `props` start to increase. I personally like this alternative approach, but completely up to you if you want to use it or not.

Reply:
sounds good to me – all about consistency 👍🏼

Reply:
You'd need this too:

```
<AgreementText agreementText ={agreement.text} />
```